src/movies.js

```diff
@@ -1,16 +1,150 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
+
+function getAllDirectors(movies){
+    return movies.map((movie) => movie.director).filter((value, index, self) => self.indexOf(value) === index);
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    return movies.filter(function(movie){
+        return (movie.director == "Steven Spielberg" && movie.genre.indexOf("Drama") !== -1);
+    }).length;
+}
+
+// 
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies){
+    return  movies.length ? Number((movies.reduce(function(acc, movie){
+        return acc + (movie.rate ? movie.rate: 0);
+    }, 0)/movies.length).toFixed(2)) : 0;
+}
+
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(movies){    
+    if (movies.length) {
+        let dramaMovies = movies.filter(movie => movie.genre[0] === 'Drama');
+        if (!dramaMovies.length) {
+            return 0;
+        }
+        else {
+            return Number(((dramaMovies.reduce((acc, filteredMovie) => {
+                return acc + filteredMovie.rate;
+             }, 0))/dramaMovies.length).toFixed(2));
+        }
+    }         
+     else {
+         return 0;
+     }
+}
+
 
-// Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(movies){
+    if (movies.length) {
+        return movies.sort(function(movie1, movie2){
```

Comment:
sort method mutates the original array, so avoid this remember to deep copy it before applying the sort method.